include/lbann/callbacks/callback.hpp

```diff
@@ -63,29 +63,19 @@ class callback_base {
   /** @name Constructors and destructor */
   ///@{
 
-  /** @brief Initialize a callback with an optional batch interval and
-   *         summarizer.
+  /** @brief Initialize a callback with an optional batch interval
    */
-  callback_base(int batch_interval = 1,
-                 lbann_summary *summarizer = nullptr) :
-    m_batch_interval(std::max(batch_interval, 1)), m_summarizer(summarizer) {}
+  callback_base(int batch_interval = 1) :
+    m_batch_interval(std::max(batch_interval, 1)) {}
   callback_base(const callback_base&) = default;
-  virtual ~callback_base() {}
+  virtual ~callback_base() = default;
 
   ///@}
   /** @name Polymorphic copy */
   ///@{
 
   virtual callback_base* copy() const = 0;
 
-  ///@}
-  /** @name Modifiers */
-  ///@{
```

Comment:
Keep this comment block or something like it. (Otherwise `setup()` will be grouped in the `Polymorphic copy` group.